cmd/cli/plugin/airgapped-network/copyimage.go

```diff
@@ -0,0 +1,229 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+
+	"sigs.k8s.io/yaml"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	tkrv1 "github.com/vmware-tanzu/tanzu-framework/apis/run/pkg/tkr/v1"
+)
+
+type imageCopyOptions struct {
+	customImageRepo                 string
+	tkgImageRepo                    string
+	tkgVersion                      string
+	customImageRepoCertificate      string
+	customTkgCompatibilityImagePath string
+	ignoreCompatibilityImage        bool
+}
+
+var totalImgCopiedCounter int
+var lc = &imageCopyOptions{}
+
+var imageCopyCmd = &cobra.Command{
+	Use:          "copyimage",
+	Short:        "Copy images to airgapped network",
+	RunE:         copyimages,
+	SilenceUsage: true,
+}
+
+func init() {
+	imageCopyCmd.Flags().StringVarP(&lc.customImageRepo, "custom-image-repo", "", "", "custom images repository for airgapped network")
+	imageCopyCmd.Flags().StringVarP(&lc.tkgImageRepo, "tkg-image-repo", "", "", "TKG public repository")
+	imageCopyCmd.Flags().StringVarP(&lc.tkgVersion, "tkg-version", "", "", "TKG version")
+	imageCopyCmd.Flags().StringVarP(&lc.customImageRepoCertificate, "custom-imageRepo-certificate", "", "", "customImageRepoCertificate")
+}
+
+func handleTkgCompatibilityImage() {
+	fmt.Printf("--- start the process of tkg-compatibility ---")
+	tkgCompatibilityRelativeImagePath := "tkg-compatibility"
+	if !IsTagRTM(lc.tkgVersion) {
+		tkgCompatibilityRelativeImagePath = filepath.Join(lc.tkgVersion, tkgCompatibilityRelativeImagePath)
+	}
+	if lc.customTkgCompatibilityImagePath != "" {
+		tkgCompatibilityRelativeImagePath = lc.customTkgCompatibilityImagePath
+	}
+	tkgCompatibilityImagePath := path.Join(lc.tkgImageRepo, tkgCompatibilityRelativeImagePath)
+	imageTags := imgpkgTagListImage(tkgCompatibilityImagePath)
+	sort.SliceStable(imageTags, func(i, j int) bool {
+		vi, err := strconv.Atoi(strings.TrimPrefix(imageTags[i], "v"))
+		if err != nil {
+			printErrorAndExit(errors.Wrapf(err, "parse tkg-compatibility image tag %s failed", imageTags[i]))
+		}
+		vj, err := strconv.Atoi(strings.TrimPrefix(imageTags[j], "v"))
+		if err != nil {
+			printErrorAndExit(errors.Wrapf(err, "parse tkg-compatibility image tag %s failed", imageTags[j]))
+		}
+		return vi < vj
+	})
+	// imgpkg copy this tkg-compatibility image
+	if lc.ignoreCompatibilityImage {
+		fmt.Printf("IGNORE_COMPATIBILITY_IMAGE is set to true, will not copy the tkg-compatibility image to airgap registry \n")
+	} else {
+		sourceImageName := tkgCompatibilityImagePath + ":" + imageTags[len(imageTags)-1]
+		destImageRepo := filepath.Join(lc.customImageRepo, tkgCompatibilityRelativeImagePath)
+		imgpkgCopyImage(sourceImageName, destImageRepo)
```

Comment:
This beats the purpose of the airgapped plugins. This assumes that the bootstrap machine has access to public internet and private network.